source/exe/main_common.h

```diff
@@ -34,8 +34,8 @@ class MainCommonBase {
 public:
   // Consumer must guarantee that all passed references are alive until this object is
   // destructed.
-  MainCommonBase(const OptionsImpl& options, Event::TimeSystem& time_system, TestHooks& test_hooks,
-                 Server::ComponentFactory& component_factory,
+  MainCommonBase(const OptionsImpl& options, Event::TimeSystem& time_system,
+                 ListenerHooks& test_hooks, Server::ComponentFactory& component_factory,
```

Comment:
This was pre-existing (I don't think I was in the review) but this might be IMO a little smoother to integrate if caller didn't have to specify these in the constructor, but could instead call addHook(...) or something.

Reply:
I don't offhand see how to implement that without either moving the server_ construction into the  run() method (which I don't like since it means you can't poke at the constructed server before you run it) or completely refactoring how hooks are done through ServerImpl (since Test/ListenerHooks is an argument to the Server::InstanceImpl constructor).  Given that I think I'll say I'm not doing this, and if you poke holes in my reasoning I'll do it in a later CL.  But I'll respond now to give you as much chance as I can to argue with me :-}.

Reply:
I don't feel that strongly about it, now that it's in.

I would've suggested plumbing that addHook() functionality thru to the ServerImpl if I had been reviewing when this was initially added, but it's not the most important thing on our list of priorities at this point.